### ISSUE TITLE
2.0.0 - Indentation error in CTE

### ISSUE THREAD

**Issue by fredriv** [sqlfluff/sqlfluff#4540]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff v2.0.0 gives an LT02 indentation error for a select statement inside a CTE which does not produce the same error with the "bare" select statement.

### Expected Behaviour

I expected the CTE to parse without errors. The same SQL parses without errors using SQLFluff v1.4.5.

### Observed Behaviour

SQLFluff gives the following linting errors:
```
L:   5 | P:  12 | LT02 | Expected line break and indent of 8 spaces before
                       | 'row_number'. [layout.indent]
L:   6 | P:   1 | LT02 | Expected indent of 12 spaces.
                       | [layout.indent]
L:   7 | P:   1 | LT02 | Expected indent of 8 spaces.
                       | [layout.indent]
```

### How to reproduce

SQL to reproduce:

```
-- Gives LT02 error when inside a CTE
with cte as (
    select a
    from b
    qualify row_number() over (
        partition by a
    ) = 1
)

select * from cte;

-- No error for the same select outside of CTE
select a
from b
qualify row_number() over (
    partition by a
) = 1;
```

### Dialect

`snowflake`

### Version

```
> sqlfluff --version
sqlfluff, version 2.0.0

> python --version
Python 3.9.9
```

### Configuration

```
[sqlfluff]
dialect = snowflake
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Good find. This I think shows a gap in the logic applied in #4420. There's a special case for bracket handling which I think should also apply to this situation (but currently doesn't). I'll look into this one in more detail.
